clientconn.go

```diff
@@ -414,6 +432,9 @@ func (cc *ClientConn) newAddrConn(addr Address, skipWait bool) error {
 		go func() {
 			if err := ac.resetTransport(false); err != nil {
 				grpclog.Printf("Failed to dial %s: %v; please retry.", ac.addr.Addr, err)
+				ac.cc.mu.Lock()
```

Comment:
Yes, this can be moved inside `resetTransport`. And actually, this `tearDown()` is redundant now, because the only error returned by `resetTransport` is `errConnClosing`.
The reason I don't want to remove this is that `resetTransport` may also return fatal connection errors like in #768, bad certificates error, and we may need to do different things based on the error and the place this function is called. Keeping it outside makes it easier. If we can make sure we always do the same thing here, this can be moved into `resetTransport`.